tests/sql/operators/test_load_file.py

```diff
@@ -715,3 +717,98 @@ def test_load_file_should_fail_loudly(sample_dag, invalid_path, caplog):
         test_utils.run_dag(sample_dag)
     expected_error = f"File(s) not found for path/pattern '{invalid_path}'"
     assert expected_error in caplog.text
+
+
+@pytest.mark.parametrize(
+    "remote_files_fixture",
+    [{"provider": "google"}],
+    indirect=True,
+    ids=["google_gcs"],
+)
+@pytest.mark.parametrize(
+    "database_table_fixture",
+    [
+        {
+            "database": Database.BIGQUERY,
+        }
+    ],
+    indirect=True,
+    ids=["bigquery"],
+)
+def test_aql_load_file_optimized_path_method_called(
+    sample_dag, database_table_fixture, remote_files_fixture
+):
+    """
+    Verify the correct method is getting called for specific source and destination.
+    """
+    db, test_table = database_table_fixture
+    file_uri = remote_files_fixture[0]
+
+    # (source, destination) : method_path - where source is file source path and destination is database
+    # and method_path is the path to method
+    optimised_path_to_method = {
+        (
+            "gs",
+            "bigquery",
+        ): "astro.databases.google.bigquery.BigqueryDatabase.gs_to_bigquery"
+    }
+
+    source = file_uri.split(":")[0]
+    destination = db.sql_type
+    mock_path = optimised_path_to_method[(source, destination)]
+
+    with mock.patch(mock_path) as gs_to_bigquery:
```

Comment:
If you want to mock the path, why not call the task directly instead of trying to run the DAG?

Reply:
Done.